tests/cache/conftest.py

```diff
@@ -0,0 +1,31 @@
+"""
+Ensure that cached functions are invalidated between test runs.
+"""
+import pytest
+
+from ddtrace.utils import cache
+
+
+_CACHED_FUNCTIONS = []
+
+old_cached = cache.cached
+
+
+def wrapped_cached(maxsize=256):
+    def wrapped_cached_f(f):
+        cached_f = old_cached(maxsize)(f)
+        _CACHED_FUNCTIONS.append(cached_f)
+        return cached_f
+
+    return wrapped_cached_f
+
+
+cache.cached = wrapped_cached
```

Comment:
Not a big fan of this but I don't really see a way around it since we have a bunch of global state in the library already :/